src/gossip/service.zig

```diff
@@ -58,43 +59,41 @@ const getWallclockMs = sig.gossip.data.getWallclockMs;
 const PACKET_DATA_SIZE = sig.net.packet.PACKET_DATA_SIZE;
 const UNIQUE_PUBKEY_CAPACITY = sig.gossip.table.UNIQUE_PUBKEY_CAPACITY;
 const MAX_NUM_PULL_REQUESTS = sig.gossip.pull_request.MAX_NUM_PULL_REQUESTS;
-const MAX_BLOOM_SIZE = sig.gossip.pull_request.MAX_BLOOM_SIZE;
 
 const PacketBatch = ArrayList(Packet);
 const GossipMessageWithEndpoint = struct { from_endpoint: EndPoint, message: GossipMessage };
 
-pub const GOSSIP_PULL_RATE_MS: u64 = 5 * std.time.ms_per_s;
-pub const GOSSIP_PULL_TIMEOUT_MS: u64 = 15 * std.time.ms_per_s;
-pub const GOSSIP_PUSH_MSG_TIMEOUT_MS: u64 = 30 * std.time.ms_per_s;
-pub const GOSSIP_PRUNE_MSG_TIMEOUT_MS: u64 = 500;
+pub const PULL_REQUEST_RATE = Duration.fromSecs(5);
+pub const PULL_RESPONSE_TIMEOUT = Duration.fromSecs(5);
+pub const ACTIVE_SET_REFRESH_RATE = Duration.fromSecs(15);
+pub const TABLE_TRIM_RATE = Duration.fromSecs(10);
+pub const BUILD_MESSAGE_LOOP_MIN = Duration.fromSecs(1);
+pub const PUBLISH_STATS_INTERVAL = Duration.fromSecs(2);
 
-pub const FAILED_INSERTS_RETENTION_MS: u64 = 20_000;
+pub const PUSH_MSG_TIMEOUT = Duration.fromSecs(30);
+pub const PRUNE_MSG_TIMEOUT = Duration.fromMillis(500);
+pub const FAILED_INSERTS_RETENTION = Duration.fromSecs(20);
+pub const PURGED_RETENTION = Duration.fromSecs(PULL_REQUEST_RATE.asSecs() * 5);
 
 pub const MAX_PACKETS_PER_PUSH: usize = 64;
 pub const MAX_BYTES_PER_PUSH: u64 = PACKET_DATA_SIZE * @as(u64, MAX_PACKETS_PER_PUSH);
-
 // 4 (enum) + 32 (pubkey) + 8 (len) = 44
 pub const MAX_PUSH_MESSAGE_PAYLOAD_SIZE: usize = PACKET_DATA_SIZE - 44;
 
-pub const GOSSIP_SLEEP_MILLIS: u64 = 100;
-pub const GOSSIP_PING_CACHE_CAPACITY: usize = 65_536;
-pub const GOSSIP_PING_CACHE_TTL_NS: u64 = std.time.ns_per_s * 1280;
-pub const GOSSIP_PING_CACHE_RATE_LIMIT_DELAY_NS: u64 = std.time.ns_per_s * (1280 / 64);
-
-pub const MAX_NUM_VALUES_PULL_RESPONSE = 20; // TODO: this is approx the rust one -- should tune
-
+pub const MAX_NUM_VALUES_PER_PULL_RESPONSE = 20; // TODO: this is approx the rust one -- should tune
+pub const NUM_ACTIVE_SET_ENTRIES: usize = 25;
 /// Maximum number of origin nodes that a PruneData may contain, such that the
 /// serialized size of the PruneMessage stays below PACKET_DATA_SIZE.
 pub const MAX_PRUNE_DATA_NODES: usize = 32;
-pub const NUM_ACTIVE_SET_ENTRIES: usize = 25;
+
+pub const PING_CACHE_CAPACITY: usize = 65_536;
+pub const PING_CACHE_TTL_NS: u64 = std.time.ns_per_s * 1280;
+pub const PING_CACHE_RATE_LIMIT_DELAY_NS: u64 = std.time.ns_per_s * (1280 / 64);
```

Comment:
Is there a reason we are not using Duration here as well?

Reply:
great point 

cc1ed30